feat: make vulnerable dependency checker pluggable

zvigrinberg opened the pull request.

ruromero commented on configs/schemas/config.schema.json:
```diff
@@ -312,6 +312,16 @@
           "type": "boolean"
+        },
+        "use_dependency_tool": {
+          "default": false,
```
The default behaviour should be the current one. Use dependency tool = True

ruromero commented on configs/from_http.json:
```diff
@@ -58,7 +58,9 @@
-    "use_uvloop": true
+    "use_uvloop": true,
+    "use_dependency_tool": false,
+    "assume_vulnerable_package_present": true
```
I think you should assume this always when the `use_dependency_tool` is `False` and never when `True`. Is there any other use case I don't see?

zvigrinberg replied:
No, just a future thinking for extension point.
as agreed, in the meantime, we will not use it.

ruromero reviewed (changes requested):
The solution is clear and simple. I just added one small remark and a question I have.
Thanks

ashsong-nv commented on configs/schemas/config.schema.json:
```diff
@@ -312,6 +312,11 @@
           "title": "Code Search Tool",
           "type": "boolean"
+        },
+        "use_dependency_tool": {
```
Could we use the name `use_dependency_checker` instead? This aligns better with `VulnerableDependencyChecker`, and avoids any confusion with agentic tools since this step occurs outside the agent.

zvigrinberg replied:
Done.

ashsong-nv reviewed (changes requested):
Thank you for making this improvement! The solution looks great to me, just had a couple of comments on implementation.

zvigrinberg commented:
> Thank you for making this improvement! The solution looks great to me, just had a couple of comments on implementation.

@ashsong-nv Thank you very much for your useful feedback!.
I Have addressed your suggestions in my recent changes.

zvigrinberg commented on configs/schemas/config.schema.json:
```diff
@@ -312,6 +312,11 @@
+        "use_dependency_tool": {
+          "default": true,
+          "title": "Whether to use VulnerableDependencyChecker tool in the pipeline or not",
+          "type": "boolean"
```
Thanks, I've Fixed it. 
For some reason the field was omitted by mistake from the class definition when i did changes .

ashsong-nv commented on configs/schemas/config.schema.json:
```diff
@@ -312,6 +312,11 @@
+        },
+        "use_dependency_checker": {
+          "default": true,
+          "title": "Whether to use VulnerableDependencyChecker tool in the pipeline or not",
```
Minor request to update the schema file to the following auto-generated version:

```suggestion
          "description": "Whether to use the VulnerableDependencyChecker in the pipeline or not.",
          "title": "Use Dependency Checker",
```

Sorry, I realized the command I shared earlier had the incorrect `--output_file` path. I just made this [fix](https://github.com/NVIDIA-AI-Blueprints/vulnerability-analysis/commit/8eb19f57d03d717210fe9e68cdd64533b6259c24) so that the default command without any output file will save to the correct location going forward.

```
python src/main.py cve tools gen-config-schema
```

ashsong-nv reviewed (approved):
One last minor suggestion, otherwise LGTM!
